models/POP/shell_scripts/cesm2_1/CESM2_1_setup_hybrid

```diff
@@ -0,0 +1,766 @@
+#                 and other useful things should be kept elsewhere.
+# RUNDIR          Will create the CESM run directory here.  Will need large
+#                 amounts of disk space, generally on a scratch partition.
+# EXEROOT         Will create the CESM executable directory here, where the
+#                 CESM executables will be built.  Medium amount of space
+#                 needed, generally on a scratch partition.
+# ARCHDIR         Will create the CESM short-term archive directories here.
+#                 Large, generally on a scratch partition.  Files will remain
+#                 here until the long-term archiver moves it to permanent storage.
+# ==============================================================================
+
+setenv CASEROOT     /glade/work/${USER}/cases/${CASE}
+setenv RUNDIR       /glade/scratch/${USER}/${CASE}/run
+setenv EXEROOT      /glade/scratch/${USER}/${CASE}/bld
+setenv ARCHDIR      /glade/scratch/${USER}/archive/${CASE}
+
+# ==============================================================================
+# Create the case - this creates the CASEROOT directory.
+#
+# For list of the pre-defined component sets: ./create_newcase --list
+# To create a variant compset, see the CESM documentation and carefully
+# incorporate any needed changes into this script.
+#
+# The script won't allow the user to make CASEROOT the same dir as where this
+# setup script is since the build process removes all files in the ${CASEROOT}
+# dir before populating it.
+# ==============================================================================
+
+if ( $CASEROOT == `dirname $0` ) then
+   echo "ERROR: the setup script should not be located in the CASEROOT"
+   echo "directory, because all files in the CASEROOT dir will be removed"
+   echo "before creating the new case.  move the script to a safer place."
+   exit -1
+endif
+
+echo "removing old files from ${CASEROOT}"
+echo "removing old files from ${EXEROOT}"
+echo "removing old files from ${RUNDIR}"
+${REMOVE} ${CASEROOT}
+${REMOVE} ${EXEROOT}
+${REMOVE} ${RUNDIR}
+
+# ==============================================================================
+# Invoking create_newcase.
+# The --run-unsupported option is necessary for CESM2 because this compset
+# hasn't been tested.
+# ==============================================================================
+
+set use_tasks_per_node = 36
+set nthreads = 1
+
+echo ''
+echo 'Invoking create_newcase.'
+echo ''
+echo '${cesmroot}/cime/scripts/create_newcase'
+echo '--case ${CASEROOT}'
+echo '--mach ${mach}'
+echo '--res ${resolution}'
+echo '--compset ${compset}'
+echo '--run-unsupported'
+echo '--pecount ${use_tasks_per_node}x${nthreads}'
+echo '--ninst ${num_instances}'
+echo '--multi-driver'
+
+${cesmroot}/cime/scripts/create_newcase \
+--case ${CASEROOT} \
+--mach ${mach} \
+--res ${resolution} \
+--compset ${compset} \
+--run-unsupported \
+--pecount ${use_tasks_per_node}x${nthreads} \
```

Comment:
I have no idea what this does.
It does not appear to impact the compilation time - and we manually clobber the pe setup with xmlchanges ... what does this do?

Reply:
I have confirmed that this only impacts the env*xml variables that we modify later.
I would prefer to remove the --pecount option from the create_newcase. I don't think it
serves a purpose and might confuse someone with regard to the xmlchanges on lines 256 to 264 below.
My first thought was that it was going to somehow adjust something for the build environment - that was apparently wrong.